library-source/fp/src/main/kotlin/si/dime/kotlin/tutorials/rest/booklibrary/librarysource/fp/Catch.kt

```diff
@@ -0,0 +1,10 @@
+package si.dime.kotlin.tutorials.rest.booklibrary.librarysource.fp
+
+import arrow.core.Either
+import arrow.core.left
+
+fun <R> catch(execution: () -> Either<Throwable, R>): Either<Throwable, R> = try {
```

Comment:
Do you understand what is FP (functional programming base idea) and what does these lines of code?

Reply:
yes, depending the result .left throws an exception and .right return the object